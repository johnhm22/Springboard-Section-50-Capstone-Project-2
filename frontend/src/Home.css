.Home {
    text-align: center;
  }

.container-fluid {
  background-color: rgba(240,248,255, 0.5);
}