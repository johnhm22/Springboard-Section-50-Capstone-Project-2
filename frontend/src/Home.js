import React, {useContext} from 'react';
import'./Home.css';
import UserContext from "./userContext";

const Home = () => {
    const user = useContext(UserContext);

    

    if(user){
        console.log("user in Home", user);
        return(
            <div className="Home">
            <div className="container-fluid text-center">
            <h1 className="b-4 font-weight-bold">Welcome to ManageMyProperty</h1>
            <h2>All the tools in one convenient place!</h2>
            <h4>Hello {user.firstname}</h4>
            <h5>Your properties are: 
            <span> {user.first_property} </span>
            <span> {user.second_property} </span>
            <span> {user.third_property} </span>
            </h5>
            {/* <p>Are you an admin?
            {user.is_admin ? <span>  Yes</span>:<span>  No</span>}</p> */}
            </div>
            </div>
            )
        }
        return(
            <div className="Home mb-2">
                <div className="container-fluid text-center pb-4">
                    <h1 className="b-4 font-weight-bold">Welcome to ManageMyProperty</h1>
                    <h2>All the tools in one convenient place!</h2>
                </div>
            </div>
        )
   
}

export default Home;