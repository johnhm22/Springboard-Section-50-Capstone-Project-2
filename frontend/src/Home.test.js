import React from 'react';
import { render } from '@testing-library/react';
import Home from './Home';

it('renders without crashing', () => {
  render(<Home />);
});

it('matches snapshot', () => {
    const {asFragment} = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
});

it('finds text that should be there', () => {
    const {queryByText} = render(<Home />);
    expect(queryByText("All the tools in one convenient place!")).toBeInTheDocument();
});


// it('show text when user is true', () => {
//     const {getByText} = render(<Home user="john" />);
//     console.log(getByText("Checking test"))
// });


