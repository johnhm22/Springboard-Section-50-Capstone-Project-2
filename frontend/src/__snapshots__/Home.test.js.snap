// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <div
      class="container-fluid text-center"
    >
      <h1
        class="b-4 font-weight-bold"
      >
        Welcome to ManageMyProperty
      </h1>
      <h2>
        All the tools in one convenient place!
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
